centreon/src/Core/HostTemplate/Application/UseCase/PartialUpdateHostTemplate/PartialUpdateHostTemplate.php

```diff
@@ -308,4 +333,64 @@ private function findOriginalMacros(HostTemplate $hostTemplate): array
 
         return [$directMacros, $inheritedMacros, $commandMacros];
     }
+
+    private function updateCategories(PartialUpdateHostTemplateRequest $request, HostTemplate $hostTemplate): void
+    {
+        $this->info(
+            'PartialUpdateHostTemplate: update categories',
+            ['host_template_id' => $hostTemplate->getId(), 'categories' => $request->categories]
+        );
+
+        if ($request->categories instanceOf NoValue) {
+            $this->info('Categories not provided, nothing to update');
+
+            return;
+        }
+
+        $categoryIds = array_unique($request->categories);
+
+        $this->assertAreValidCategories($categoryIds);
+
+        if ($this->user->isAdmin()) {
+            $originalCategories = $this->readHostCategoryRepository->findByHost($hostTemplate->getId());
+        } else {
+            $originalCategories = $this->readHostCategoryRepository->findByHostAndAccessGroups(
+                $hostTemplate->getId(),
+                $this->accessGroups
+            );
+        }
+
+        $originalCategoryIds = array_map((fn($category) => $category->getId()), $originalCategories);
+
+        $categoryDiff = new BasicDifference($originalCategoryIds, $categoryIds);
+        /** @var int[] $addedCategories */
+        $addedCategories = $categoryDiff->getAdded();
+        /** @var int[] $removedCategories */
+        $removedCategories = $categoryDiff->getRemoved();
```

Comment:
Theorically not needed phpdoc.
I looked at `BasicDifference` and forgot to make it more generic.
👉🏻  Please contact me to see together how to fix the `BasicDifference` phpdoc.